openapi3filter/validate_request_test.go

```diff
@@ -107,26 +107,4 @@ func Example() {
 		fmt.Println(err)
 	}
 	// Output:
-	// request body has an error: doesn't match the schema: Doesn't match schema "oneOf"
-	// Schema:
-	//   {
-	//     "discriminator": {
-	//       "propertyName": "pet_type"
-	//     },
-	//     "oneOf": [
-	//       {
-	//         "$ref": "#/components/schemas/Cat"
-	//       },
-	//       {
-	//         "$ref": "#/components/schemas/Dog"
-	//       }
-	//     ]
-	//   }
-	//
-	// Value:
-	//   {
-	//     "bark": true,
-	//     "breed": "Dingo",
-	//     "pet_type": "Cat"
-	//   }
```

Comment:
I understand the request data should be invalid per the provided schema. What am I missing?

The test used to not pass because there was no output, because no error is returned. Are you sure this should not output anything?

Reply:
This relates to your original question about not being sure regarding the presence of the "`else`" statement:

Quoting from https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/

>"It is implied, that the property to which discriminator refers, contains the name of the target schema. In the example above, the objectType property should contain either simpleObject, or complexObject string. If the property values do not match the schema names, you can map the values to the names. To do this, use the discriminator/mapping keyword:"

I'm interpreting this as saying that if you don't specify the mapping the default mapping is based on the names of the objects included in the oneOf list: `simpleObject`, `complexObject` in the spec doc, `Dog` or `Cat` in the `Example()` from `validate_request_test.go`.

Let me know if you believe I'm getting it all wrong or it makes sense.